azure/services/managedclusters/spec_test.go

```diff
@@ -163,6 +165,67 @@ func TestParameters(t *testing.T) {
 				g.Expect(result).To(BeNil())
 			},
 		},
+		{
+			name:     "set Linux profile if SSH key is set",
+			existing: nil,
+			spec: &ManagedClusterSpec{
+				Name:            "test-managedcluster",
+				ResourceGroup:   "test-rg",
+				Location:        "test-location",
+				Tags:            nil,
+				Version:         "v1.22.0",
+				LoadBalancerSKU: "Standard",
+				SSHPublicKey:    base64.StdEncoding.EncodeToString([]byte("test-ssh-key")),
+				GetAllAgentPools: func() ([]azure.ResourceSpecGetter, error) {
+					return []azure.ResourceSpecGetter{
+						&agentpools.AgentPoolSpec{
+							Name:          "test-agentpool-0",
+							Mode:          string(infrav1.NodePoolModeSystem),
+							ResourceGroup: "test-rg",
+							Replicas:      int32(2),
+							AdditionalTags: map[string]string{
+								"test-tag": "test-value",
+							},
+						},
+					}, nil
+				},
+			},
+			expect: func(g *WithT, result interface{}) {
+				g.Expect(result).To(BeAssignableToTypeOf(containerservice.ManagedCluster{}))
+				g.Expect(result.(containerservice.ManagedCluster).LinuxProfile).To(Not(BeNil()))
+				g.Expect(result.(containerservice.ManagedCluster).LinuxProfile.SSH.PublicKeys).To(Equal(base64.StdEncoding.EncodeToString([]byte("test-ssh-key"))))
```

Comment:
I apologize @maciaszczykm , my suggestion seems to have broken your test. 
I wanted to suggest testing the contents of `result.(containerservice.ManagedCluster).LinuxProfile.SSH.PublicKeys` rather than just checking _it not to be nil_ for higher confidence.
This is what the test complains about
```
Expected
                <*[]containerservice.SSHPublicKey | 0xc000012a98>: [
                    {
                        KeyData: "test-ssh-key",
                    },
                ]
            to equal
                <string>: dGVzdC1zc2gta2V5
```
We can either compare the value of `result.(containerservice.ManagedCluster).LinuxProfile.SSH.PublicKeys[0].KeyData` with `"test-ssh-key"` or revert the change, essentially removing the line `g.Expect(result.(containerservice.ManagedCluster).LinuxProfile.SSH.PublicKeys).To(Equal(base64.StdEncoding.EncodeToString([]byte("test-ssh-key"))))` .

Sorry to cause such a back-and-forth on this.

Reply:
Sure, no worries. I will update it in a moment.

Reply:
It's done.